### PR TITLE
Add TargetFrameworkTestHostLoadBalancingStrategy multihost strategy

### PR DESCRIPTION
**Pull request by MarcoRossignoli**
Add MultiHostTestExecution

### CODE REVIEW

File: playground/TestPlatform.Playground/Program.cs [hunk 1]
```diff
@@ -44,9 +44,20 @@ static void Main()
...
                     <!-- <MaxCpuCount>1</MaxCpuCount> -->
+                    <!-- <DisableParallelization>True<DisableParallelization> -->
                     <!-- <TargetPlatform>x86</TargetPlatform> -->
                     <!-- <TargetFrameworkVersion>net472</TargetFrameworkVersion> -->
 
+                    <!-- Per test coverage support -->
+                    <!--
+                    <MaxCpuCount>0</MaxCpuCount>
+                    <ForceOneTestAtTimePerTestHost>True</ForceOneTestAtTimePerTestHost>
+                    <MultiHostTestExecution strategy="Fixed">
```
[the remark this answers is not in the document]
Comment (MarcoRossignoli) [reply playground/TestPlatform.Playground/Program.cs:55]: I'll simplify

[the remark this answers is not in the document]
Comment (MarcoRossignoli) [reply playground/TestPlatform.Playground/Program.cs:55]: it's fine to me

File: src/Microsoft.TestPlatform.CrossPlatEngine/Client/Parallel/ParallelProxyExecutionManager.cs [hunk 2]
```diff
@@ -257,6 +262,30 @@ private List<ProviderSpecificWorkload<TestRunCriteria>> SplitToWorkloads(TestRun
                     // Create multiple testcase batches, each having set of testcases from single source,
                     // so each testhost will end up running one source.
                     testCaseBatches = group.Select(w => sourceToTestCasesMap[w.Work]).ToList();
+                    if (numberOfTestHostToUse is not null)
+                    {
+                        if (numberOfTestHostToUse > _parallelLevel)
+                        {
+                            EqtTrace.Info($"ParallelProxyExecutionManager: adjust the numberOfTestHostToUse to the max parallel level, from {numberOfTestHostToUse} to {_parallelLevel}");
```
[the remark this answers is not in the document]
Comment (MarcoRossignoli) [reply src/Microsoft.TestPlatform.CrossPlatEngine/Client/Parallel/ParallelProxyExecutionManager.cs:269]: We could, issue is that for not is usable only in design mode so it won't be visible but in the log, I'll try to understand if we have the message logger somewhere to send back to VS.

[the remark this answers is not in the document]
Comment (MarcoRossignoli) [reply src/Microsoft.TestPlatform.CrossPlatEngine/Client/Parallel/ParallelProxyExecutionManager.cs:269]: No way we don't have the logger, moved to warning log for now

File: src/Microsoft.TestPlatform.CrossPlatEngine/TestEngine.cs [hunk 3]
```diff
@@ -692,4 +698,34 @@ private static void WarnAboutNotFoundRuntimeProvidersOrThrowWhenNoneAreFound(Lis
             }
         }
     }
+
+    /// <summary>
+    /// We don't add this helper to the XmlRunSettingsUtilities because the feature is in preview and not exposed yet
+    /// </summary>
+    internal static int? GetMultiHostTestExecutionHostCount(string? runsettings)
```
Comment (Evangelink) [inline src/Microsoft.TestPlatform.CrossPlatEngine/TestEngine.cs:705]: Maybe you want to rename this method.

File: playground/TestPlatform.Playground/Program.cs [hunk 4]
```diff
@@ -44,9 +44,17 @@ static void Main()
...
                     <!-- <MaxCpuCount>1</MaxCpuCount> -->
+                    <!-- <DisableParallelization>True<DisableParallelization> -->
                     <!-- <TargetPlatform>x86</TargetPlatform> -->
                     <!-- <TargetFrameworkVersion>net472</TargetFrameworkVersion> -->
 
+                    <!-- Per test coverage support -->
+                    <!--
+                    <MaxCpuCount>0</MaxCpuCount>
+                    <ForceOneTestAtTimePerTestHost>True</ForceOneTestAtTimePerTestHost>
+                    <TargetFrameworkTestHostLoadBalancingStrategy>4</TargetFrameworkTestHostLoadBalancingStrategy>
```
Comment (Evangelink) [inline playground/TestPlatform.Playground/Program.cs:55]: Woopsie :) This naming suggestion was for keeping your strategy idea. 

Hum, I don't have super good name popping up in my mind... Maybe your multi-host suggestion was not bad... Ideas: `SpecificMultiTestHostCount`, `TargetFrameworkTestHostDemultiplexer`?

**Review by Evangelink — APPROVED** [review]

**Pull request merged**